pkg/clusteroperator/clusteroperator.go

```diff
@@ -0,0 +1,172 @@
+)
+
+const (
+	fiveScds time.Duration = 5 * time.Second
+	isTrue                 = "True"
+)
+
+// Builder provides struct for clusterOperator object.
+type Builder struct {
+	// ClusterOperator definition. Used to create a clusterOperator object.
+	Definition *v1.ClusterOperator
+	// Created clusterOperator object.
+	Object *v1.ClusterOperator
+	// apiClient opens api connection to the cluster.
+	apiClient *clients.Settings
+	// Used in functions that define or mutate clusterOperator definition. errorMsg is processed before the
+	// ClusterOperator object is created.
+	errorMsg string
+}
+
+// Exists checks whether the given clusterOperator exists.
+func (builder *Builder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if clusterOperator %s exists", builder.Definition.Name)
+
+	_, err := builder.apiClient.ClusterOperators().Get(
+		context.Background(),
+		builder.Definition.Name,
+		metaV1.GetOptions{})
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// IsAvailable check if the clusterOperator is Available.
+func (builder *Builder) IsAvailable() bool {
+	glog.V(100).Infof("Verify %s clusterOperator availability", builder.Definition.Name)
+
+	if !builder.Exists() {
+		return false
+	}
+
+	for _, condition := range builder.Object.Status.Conditions {
+		if condition.Type == "Available" {
+			return condition.Status == isTrue
+		}
+	}
+
+	return false
+}
+
+// IsDegraded check if the clusterOperator is Degraded.
+func (builder *Builder) IsDegraded() bool {
+	glog.V(100).Infof("Verify %s clusterOperator is degraded", builder.Definition.Name)
+
+	if !builder.Exists() {
+		return false
+	}
+
+	for _, condition := range builder.Object.Status.Conditions {
+		if condition.Type == "Degraded" {
+			return condition.Status == isTrue
+		}
+	}
+
+	return false
+}
+
+// IsProgressing check if the clusterOperator is Degraded.
+func (builder *Builder) IsProgressing() bool {
+	glog.V(100).Infof("Verify %s clusterOperator is progressing", builder.Definition.Name)
+
+	if !builder.Exists() {
+		return false
+	}
+
+	for _, condition := range builder.Object.Status.Conditions {
+		if condition.Type == "Progressing" {
+			return condition.Status == isTrue
+		}
+	}
+
+	return false
+}
+
+// WaitUntilAvailable waits for timeout duration or until clusterOperator is Available.
+func (builder *Builder) WaitUntilAvailable(timeout time.Duration) error {
+	return builder.WaitUntilInStatus("Available", timeout)
+}
+
+// WaitUntilProgressing waits for timeout duration or until clusterOperator is Progressing.
+func (builder *Builder) WaitUntilProgressing(timeout time.Duration) error {
+	return builder.WaitUntilInStatus("Progressing", timeout)
+}
+
+// WaitUntilInStatus waits for timeout duration or until clusterOperator gets to a specific status.
+func (builder *Builder) WaitUntilInStatus(
+	status v1.ClusterStatusConditionType, timeout time.Duration) error {
```

Comment:
Instead of status, I think a better argument name would be conditionType. The status is actually just "True".

Reply:
Looks the parameter is still called status

Reply:
have no idea what happened, now it is fine for sure

Reply:
It is still show as status to me

Reply:
yep, it was